config/config.go

```diff
@@ -0,0 +1,106 @@
+package config
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"github.com/hashicorp/hcl"
+	"github.com/hashicorp/hil"
+	"github.com/hashicorp/hil/ast"
+)
+
+// Config is a variable file config definition
+type Config struct {
+	Variables []VariableConfig `hcl:"variable"`
+	Templates []TemplateConfig `hcl:"template"`
+}
+
+// VariableConfig defines the structure for our variable config sections
+type VariableConfig struct {
+	Name  string      `hcl:",key"`
+	Value interface{} `hcl:"value"`
+}
+
+// TemplateConfig defines the structure for our template config sections
+type TemplateConfig struct {
+	Name    string `hcl:",key"`
+	Content string `hcl:"content"`
+	Count   int    `hcl:"count"`
```

Comment:
At the moment this isn't used. I should either decide on implementing the "count" template feature, or remove this field.